src/xtgeo/grid3d/_gridprops_import_roff.py

```diff
@@ -1,65 +1,75 @@
-from typing import List, Union
+from __future__ import annotations
 
-from typing_extensions import Literal
+import io
+from typing import TYPE_CHECKING, List, Literal, Union
+
+import roffio
 
 import xtgeo
-from xtgeo.common.sys import _XTGeoFile
+from xtgeo.common import XTGeoDialog
 
-from . import _grid3d_utils as utils
-from .grid_property import GridProperty
+if TYPE_CHECKING:
+    from xtgeo.common.sys import _XTGeoFile
 
-xtg = xtgeo.common.XTGeoDialog()
+    from .grid_property import GridProperty
 
-logger = xtg.functionlogger(__name__)
+logger = XTGeoDialog().functionlogger(__name__)
 
 
 def import_roff_gridproperties(
     pfile: _XTGeoFile,
     names: Union[List[str], Literal["all"]],
     strict: bool = True,
 ) -> List[GridProperty]:
-    """Imports list of properties from a roff file.
+    """
+    Imports a list of properties from a ROFF file.
+
+    Parameters:
+        pfile:
+            Reference to the file.
+        names:
+            List of names to fetch, can also be "all" to fetch all properties.
+        strict:
+            If strict=True, will raise error if key is not found.
+            Defaults to True.
 
-    Args:
-        pfile: Reference to the file
-        names: List of names to fetch, can also be "all" to fetch all properties.
-        strict: If strict=True, will raise error if key is not found.
     Returns:
         List of GridProperty objects fetched from the ROFF file.
+
     """
-    print(pfile)
     validnames = []
+    with roffio.lazy_read(pfile.file) as contents:
+        for tagname, tagkeys in contents:
+            for keyname, values in tagkeys:
+                if tagname == "parameter" and keyname == "name":
+                    validnames.append(values)
 
-    collectdata = utils.scan_keywords(pfile, fformat="roff")
-    for item in collectdata:
-        keyname = item[0]
-        if keyname.startswith("parameter!name!"):
-            # format is 'parameter!name!FIPNUM'
-            validnames.append(keyname.split("!").pop())
+    # Rewind if this file is in memory
+    if isinstance(pfile.file, (io.BytesIO, io.StringIO)):
+        pfile.file.seek(0)
```

Comment:
Feels like something that the context manger in roffio should do?

Reply:
I can see the usefulness of both rewinding and not, not sure what is most common for streams. But currently it doesn't do this.